lib/transaction.ex

```diff
@@ -31,6 +31,67 @@ defmodule Elixium.Transaction do
     %{outputs: outputs}
   end
 
+  @doc """
+  Creates a correct tx id
+  """
+  @spec create_tx_id(List, String.t()) :: String.t()
+  def create_tx_id(tx, tx_timestamp) do
+    calculate_hash(tx) <> tx_timestamp
+    |> Utilities.sha_base16()
+  end
+
+  @doc """
+  Creates a singature list based off the input being passed in and the corresponding private key
+  """
+  @spec create_sig_list(Map, Map) :: Tuple
+  def create_sig_list(input, transaction) do
+    priv = Elixium.KeyPair.get_priv_from_file(input.addr)
+    digest = Elixium.Transaction.signing_digest(transaction)
+    sig = Elixium.KeyPair.sign(priv, digest)
+    {input.addr, sig}
+  end
+
+  @doc """
+  Take the correct amount of Utxo's to send the alloted amount in a transaction.
+  """
+  @spec take_necessary_utxos(List, Decimal) :: function
+  def take_necessary_utxos(utxos, amount), do: take_necessary_utxos(utxos, [], amount)
+
+  @spec take_necessary_utxos(List, List, Decimal) :: List
+  def take_necessary_utxos(utxos, chosen, amount) do
+    if D.cmp(amount, 0) == :gt do
+      if utxos == [] do
+        :not_enough_balance
+      else
+        [utxo | remaining] = utxos
+        take_necessary_utxos(remaining, [utxo | chosen], D.sub(amount, utxo.amount))
+      end
+    else
+      chosen
+    end
+  end
+
+  @doc """
+  Creates a current time stamp for transaction building
+  """
+  @spec create_timestamp :: String.t()
+  def create_timestamp, do: DateTime.utc_now |> DateTime.to_string
```

Comment:
Looks like this is a duplicate of `time_unix/0` in block.ex. We should probably move the function to utilities and call it from there if we're using it in multiple places

Reply:
Only difference is this pipes to DateTime.to_string, instead of DateTime.to_unix

Reply:
I don't think this function is used anymore since it was removed from the create_tx_id function, so we should be able to just remove it

Reply:
Removed & Updated Both Tests & Functions